### PR TITLE
[Yokogawa In the middle of STEP9] Introduction of I18n

### PR DESCRIPTION
**Pull request by Take-st**
## Overview
-  In the middle of  `STEP9 configuring rails application` of the [Rails training](https://github.com/Fablic/training/blob/develop/steps_jp.md)

## What I've DONE
- Introduction of  I18n
- Internationalization of the things below
  - views
  - controller

### CODE REVIEW

File: myapp/app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,10 @@
 class ApplicationController < ActionController::Base
+
+  around_action :switch_locale
+
+  def switch_locale(&action)
+    locale = params[:locale] || I18n.default_locale
+    I18n.with_locale(locale, &action)
+  end
```
Comment (Take-st) [inline myapp/app/controllers/application_controller.rb:8]: see: https://railsguides.jp/i18n.html#%E6%97%A5%E4%BB%98%E3%83%BB%E6%99%82%E5%88%BB%E3%83%95%E3%82%A9%E3%83%BC%E3%83%9E%E3%83%83%E3%83%88%E3%82%92%E8%BF%BD%E5%8A%A0%E3%81%99%E3%82%8B

File: myapp/config/application.rb [hunk 2]
```diff
@@ -18,5 +18,8 @@ class Application < Rails::Application
     #
     # config.time_zone = "Central Time (US & Canada)"
     # config.eager_load_paths << Rails.root.join("extras")
+
+    config.i18n.available_locales = [:en, :ja]
+    config.i18n.default_locale = :ja
```
Comment (Take-st) [inline myapp/config/application.rb:23]: Set the default locale Japanese here.

File: myapp/app/controllers/application_controller.rb (hunk 1 above)
Comment (Take-st) [reply myapp/app/controllers/application_controller.rb:8]: You can switch the locale by setting `?locale=en` as a query parameter.


### Japanese (http://localhost:3000/)
image

### English (http://localhost:3000/?locale=en)
image

**Comment by Take-st** [comment]
[note]

- TODO: Now once you move to another page, the query parameter for locale is reset and see japanese.
  - fix this by taking over the query parameter for locale
- [updated] I have completed the TODO above.
  - https://github.com/Fablic/training/pull/1509/commits/cf3cca253ef1f1c78b4fd996589a100563ce3a4d
  - https://github.com/Fablic/training/pull/1509/commits/cbbe8f9498eb18072d6efa62a829bda548b416ce

File: myapp/app/controllers/application_controller.rb (hunk 1 above)
Comment (DaisukeHirabayashi) [reply myapp/app/controllers/application_controller.rb:8]: にゃるほどです！

File: myapp/app/views/tasks/_form.html.erb [hunk 3]
```diff
@@ -1,11 +1,11 @@
 <%= form_with model: @task do |form| %>
-  <p>name</p>
+  <p><%= t(:name, scope: [:activerecord, :attributes, :task]) %></p>
```
Comment (DaisukeHirabayashi) [inline myapp/app/views/tasks/_form.html.erb:2]: scopeでまとめれるんですねーーー！

**Review by DaisukeHirabayashi — APPROVED** [review]
lgtm!

**Pull request merged**